src/TTextEdit.cpp

```diff
@@ -1545,3 +1545,16 @@ int TTextEdit::bufferScrollDown(int lines)
         return lines;
     }
 }
+
+int TTextEdit::getColumnCount()
+{
+    // Not using 'mFontWidth' as currently it is the with of the 'W' char and
+    // we should use the "space" width for this, as discussed on #1438
+    return width() / QFontMetrics(mpHost->mDisplayFont).width(QChar(' '));
+}
+
+int TTextEdit::getRowCount()
+{
+    return height() / mFontHeight;
```

Comment:
`mFontHeight` is set, not surprisingly, to `QFontMetrics::height()` which is always equivalent to the sum of: `QFontMetrics::ascent()` and `QFontMetrics::descent()` however `QFontMetrics::lineSpacing()` is `QFontMetrics::height()` plus `QFontMetrics::leading()` - so I am not entirely sure that this will return the right value for "the number of lines that can be fitted into the visible window" as it does not account for space _between_ the lines of text (assuming there _is_ any)!

I.e to show 10 lines of text completely you will need to divide by `lineSpacing()` to account for the `leading()` space between the lines that are `height()` tall {the `QFontMetrics` one not the `QWidget::height()` here}...!

Given that we are doing things with divisionsI wonder if we may wish to consider using `QFontMetricsF` in some places and rounding the results with `qRound(`...`)`?

Reply:
I've tested it and it works correctly - did you run into any issues with this in your tests?